Fix missing optional key warnings when AllPatternsMandatory is enabled.

StarKhan6368 opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->

**What**: Fix missing optional key warnings when AllPatternsMandatory is enabled.

<!-- Why are these changes necessary? -->

**Why**:

<!-- How were these changes implemented? -->

**How**:

<!-- Have you done all of these things?  -->

**Checklist**:

<!-- add "N/A" to the end of each line that's irrelevant to your changes -->

<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->

- [ ] Documentation added to the README.md OR link to PR on https://github.com/specmatic/specmatic-documentation
- [x] Tests
- [ ] Sonar Quality Gate

<!-- feel free to add additional comments -->

StarKhan6368 committed:
Fix AllPatternsMandatory optional param missing.

- All Optional params that are missing should be shown.

StarKhan6368 committed:
Update tests to ensure pattern seen fix

StarKhan6368 committed:
Update tests assertion methods for strings.

- isEqualToNormalizingWhitespace instead of containsIgnoringWhitespaces
  when possible, so anything additional is caught.

StarKhan6368 committed:
Add tests for AnyPattern when allPatternsMandatory

StarKhan6368 committed:
Add tests for ListPattern when allPatternsMandatory

joelrosario reviewed (changes requested):
After merging with main, the optional keys are missed once again. Please rebase/merge against main and fix this.

StarKhan6368 committed:
Fix AnyPattern missing optional warnings.

- AnyPattern should not add itself as seen as the discriminator
  mappings can ref back to itself.

StarKhan6368 committed:
Merge branch 'main' into mandatory-patterns-fixes

# Conflicts:
#	core/src/main/kotlin/io/specmatic/core/pattern/JSONObjectPattern.kt
#	core/src/test/kotlin/integration_tests/LoadTestsFromExternalisedFiles.kt
#	core/src/test/kotlin/io/specmatic/core/pattern/AnyPatternTest.kt

StarKhan6368 committed:
Fix AdditionalProperties failing tests
